### ISSUE TITLE
Error with CACHE_URL passwords that end with '='

### ISSUE THREAD

**Issue by davidfischer** [joke2k/django-environ#200]
This is somewhat similar to #194. A cache password that ends with the `=` character raises an error when calling `env.cache`. Even escaping it (`%3D`) will not work.

It looks like the `cast` method works by trying to just convert to a value and if that fails with a `ValueError`, it assumes the value is string ([see code](https://github.com/joke2k/django-environ/blob/20a1b50/environ/environ.py#L27-L30)). However, a string that ends in a `=` will throw a `SyntaxError` instead of a `ValueError`:

    >>> import ast
    >>> ast.literal_eval('mypassword=')
    Traceback (most recent call last):
    ...
    SyntaxError: unexpected EOF while parsing

If you think the solution is just to catch a `SyntaxError` in addition to a `ValueError`, I'm happy to make a PR.

Normally I'd just change the password but on Azure the generated cache passwords always end with a `=` for some reason.

**Comment by ksze** [comment]
> on Azure the generated cache passwords always end with a `=` for some reason

Is the password really just the base64 of a long array of random bytes?

**Comment by davidfischer** [comment]
Correct.

**Comment by pix666** [comment]
A few more examples for which a `SyntaxError` exception is raised by `_cast()`. Esaping does not help.
* `anything-` (minus at the end)
* `anything*` (asterisk at the end)
* `*anything` (asterisk at the start)
* `anything.` (dot at the end)
* `anything.1` (dot near digit)
* `(anything` (unbalanced braces anywhere)
* `anything-v1.2` (God knows why. For some reason `anything-1.2` works well)

I came across this when I tried to set the value of `key_prefix` for Redis.
